manifests/fragment.pp

```diff
@@ -1,41 +1,22 @@
-# == Define: concat::fragment
-#
 # Creates a concat_fragment in the catalogue
 #
-# === Options:
-#
-# [*target*]
-#   The file that these fragments belong to
-# [*content*]
-#   If present puts the content into the file
-# [*source*]
-#   If content was not specified, use the source
-# [*order*]
+# @param target The file that these fragments belong to
+# @param content If present puts the content into the file
+# @param source If content was not specified, use the source
+# @param order
 #   By default all files gets a 10_ prefix in the directory you can set it to
 #   anything else using this to influence the order of the content in the file
 #
 define concat::fragment(
-    $target,
-    $ensure  = undef,
-    $content = undef,
-    $source  = undef,
-    $order   = '10',
+  String                  $target,
+  Optional[String]        $content = undef,
+  Variant[String,Integer] $order   = 10,
```

Comment:
Good idea; I tried to do this in #437 but ran into https://tickets.puppetlabs.com/browse/PUP-7440 so the fail couldn't go away.